### PR TITLE
Migrate make_dfu_header to CLI

### PR DESCRIPTION
**Pull request by zvecr**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/python/qmk/cli/generate/dfu_header.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+@cli.argument('-q', '--quiet', arg_only=True, action='store_true', help="Quiet mode, only output error messages")
+@cli.argument('-kb', '--keyboard', help='Keyboard to generate LUFA Keyboard.h for.')
+@cli.subcommand('Used by the make system to generate LUFA Keyboard.h from info.json', hidden=True)
+@automagic_keyboard
+def generate_dfu_header(cli):
+    """Generates the Keyboard.h file.
+    """
+    # Determine our keyboard(s)
+    if not cli.config.generate_dfu_header.keyboard:
+        cli.log.error('Missing paramater: --keyboard')
```
Comment (fauxpark) [inline lib/python/qmk/cli/generate/dfu_header.py:21]:
```suggestion
        cli.log.error('Missing parameter: --keyboard')
```

File: data/mappings/info_config.json [hunk 2]
```diff
@@ -38,5 +38,9 @@
     "RGBLIGHT_SPLIT": {"info_key": "rgblight.split", "value_type": "bool"},
     "PRODUCT": {"info_key": "keyboard_folder", "to_json": false},
     "PRODUCT_ID": {"info_key": "usb.pid", "value_type": "hex"},
-    "VENDOR_ID": {"info_key": "usb.vid", "value_type": "hex"}
+    "VENDOR_ID": {"info_key": "usb.vid", "value_type": "hex"},
+    "QMK_ESC_OUTPUT": {"info_key": "qmk_dfu.esc_output"},
```
Comment (fauxpark) [inline data/mappings/info_config.json:42]: Thinking this should be less tied to DFU, as these defines could also be used for the LUFA HID bootloader.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: data/mappings/info_config.json (hunk 2 above)
Comment (zvecr) [reply data/mappings/info_config.json:42]: I originally had "bootloader" but that was already taken, will have a think as i also didnt like how it was "qmk-dfu" in one context but then "qmk_dfu" elsewhere.

Comment (fauxpark) [reply data/mappings/info_config.json:42]: Maybe `qmk_lufa_bootloader`? That could also give a hint as to the fact QMK-DFU (and potentially QMK-HID) are only for LUFA-based boards.

**Review by fauxpark — APPROVED** [review]

**Review by skullydazed — APPROVED** [review]

**Pull request merged**